apps/protailwind/src/video/exercise-overlay.tsx

```diff
@@ -594,46 +594,66 @@ const handleContinue = async ({
   path,
 }: {
   router: NextRouter
-  module: SanityDocument
-  section?: SanityDocument
-  nextExercise?: LessonResource | null
+  module: Module
+  section?: Section | null
+  nextExercise?: Lesson | null
   handlePlay: () => void
   path: string
 }) => {
   if (nextExercise?._type === 'solution') {
     if (section) {
-      const exercise = section.lessons.find((exercise: SanityDocument) => {
-        const solution = exercise.solution
-        return solution?._key === nextExercise._key
-      })
+      const exercise =
+        section.lessons &&
+        section.lessons.find((exercise: Exercise) => {
+          const solution = exercise.solution
+          return solution?._key === nextExercise._key
+        })
 
-      return await router
-        .push({
-          query: {
-            module: module.slug.current,
-            section: section.slug,
-            lesson: exercise.slug,
-          },
+      return exercise
+        ? await router
+            .push({
+              query: {
+                module: module.slug.current,
+                section: section.slug,
+                lesson: exercise.slug,
+              },
 
-          pathname: `${path}/[module]/[section]/[lesson]/solution`,
-        })
-        .then(() => handlePlay())
+              pathname: `${path}/[module]/[section]/[lesson]/solution`,
+            })
+            .then(() => handlePlay())
+        : await router.push({
+            query: {
+              module: module.slug.current,
+            },
+
+            pathname: `${path}/[module]`,
+          })
     } else {
-      const exercise = module.lessons.find((exercise: SanityDocument) => {
-        const solution = exercise.solution
-        return solution?._key === nextExercise._key
-      })
+      const exercise =
+        module.lessons &&
+        module.lessons.find((exercise: Exercise) => {
+          const solution = exercise.solution
+          return solution?._key === nextExercise._key
+        })
 
-      return await router
-        .push({
-          query: {
-            module: module.slug.current,
-            lesson: exercise.slug,
-          },
+      return exercise
+        ? await router
+            .push({
+              query: {
+                module: module.slug.current,
+                lesson: exercise.slug,
+              },
 
-          pathname: `${path}/[module]/[lesson]/solution`,
-        })
-        .then(() => handlePlay())
+              pathname: `${path}/[module]/[lesson]/solution`,
+            })
+            .then(() => handlePlay())
+        : await router.push({
+            query: {
+              module: module.slug.current,
+            },
+
+            pathname: `${path}/[module]`,
+          })
```

Comment:
this routing stuff is zany and deserves a think, and maybe consolidation in skill-lesson since structurally it will be the same. moving to "a module always has a section" probably helps the logic